adapters/marsmedia/marsmedia.go

```diff
@@ -0,0 +1,176 @@
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+	validImpExists := false
+	for i := 0; i < len(request.Imp); i++ {
+		if request.Imp[i].Banner != nil {
+			bannerCopy := *request.Imp[i].Banner
+			if bannerCopy.W == nil && bannerCopy.H == nil && len(bannerCopy.Format) > 0 {
+				firstFormat := bannerCopy.Format[0]
+				bannerCopy.W = &(firstFormat.W)
+				bannerCopy.H = &(firstFormat.H)
+			}
+			request.Imp[i].Banner = &bannerCopy
+			validImpExists = true
+		} else if request.Imp[i].Video != nil {
+			validImpExists = true
+			videoCopy := *request.Imp[i].Video
+			request.Imp[i].Video = &videoCopy
+		}
+	}
+	if !validImpExists {
+		err := &errortypes.BadInput{
+			Message: fmt.Sprintf("No valid impression in the bid request"),
+		}
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	request.AT = 1 //Defaulting to first price auction for all prebid requests
+
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+	thisURI := a.URI
+	thisURI = thisURI + "&dev=1"
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	if request.Device != nil {
+		addHeaderIfNonEmpty(headers, "User-Agent", request.Device.UA)
+		addHeaderIfNonEmpty(headers, "X-Forwarded-For", request.Device.IP)
+		addHeaderIfNonEmpty(headers, "Accept-Language", request.Device.Language)
+		if request.Device.DNT != nil {
+			addHeaderIfNonEmpty(headers, "DNT", strconv.Itoa(int(*request.Device.DNT)))
+		}
+	}
+
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     thisURI,
+		Body:    reqJSON,
+		Headers: headers,
+	}}, errors
+}
+
+func (a *MarsmediaAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: %d. ", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+
+	var bidResp openrtb.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("bad server response: %d. ", err),
+		}}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(bidResp.SeatBid[0].Bid))
+	sb := bidResp.SeatBid[0]
+	for i := 0; i < len(sb.Bid); i++ {
+		bid := sb.Bid[i]
+		bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+			Bid:     &bid,
+			BidType: "banner",//getMediaTypeForImp(bid.ImpID, internalRequest.Imp),
```

Comment:
Why are we not executing the `getMediaTypeForImp(bid.ImpID, internalRequest.Imp)`? This adapter does handle video imps, isn't it? If we decide that even video imps will get the `bidType` value of "banner" by default, can we get rid of both the comment and the `getMediaTypeForImp(bid.ImpID, internalRequest.Imp)` function alltogether?